### DISCUSSION TITLE
WiX 4.0 project incompatible with VS 17.8 on one machine but not another due to .vs directory

### DISCUSSION THREAD

**Discussion by twest820** [wixtoolset/issues#7907] [Questions]
Not sure if this is a known issue but I thought it might be worth noting in case someone else hits it. Doesn't look to me like it's a WiX problem, so I'm putting this in discussions.

**Scenario**
Migrating a .NET 8.0 solution containing a WiX 4.0 installer project from an older dev machine to a newer one. Both machines are Windows 10 2022H2 and have the same VS 17.8 Community Edition components and HeatWave for VS 2022 installed. Both successfully build all of the class libraries, executables, and unit tests in the solution. However, on the newer machine Visual Studio has the WiX project as `(incompatible)` and says `The application is not installed.` even though HeatWave installed without issues. Failure to load the WiX project is persistent through Visual Studio restarts and such.

**Fix that worked for me**
Delete the contents of Visual Studio's hidden $(SolutionDir)\.vs directory and restart Visual Studio. This is crude but I was needing to get the installer building and lacking in time to debug Visual Studio's internal state.

**Best guess at cause**
Sequence of events here was I cloned the repo to the new machine, opened the solution, and then remembered I needed to add the HeatWave extension. So Visual Studio was correct in initially marking the installer project incompatible when the solution was first opened. Not sure why VS doesn't seem to reassess subsequently, but this might be a VS tries once and decides WiX is forever unknown thing.

**Comment by chrpai** [comment]
Votive was part of WiX proper in v3 but in v4  Heatwave isn't.  Would you mind filing this over at https://github.com/firegiant/HeatWaveSupport/issues ?
